crates/pathfinder/src/state/merkle_tree.rs

```diff
@@ -1443,11 +1441,11 @@ mod tests {
             uut.set(key.view_bits(), value).unwrap();
 
             let mut visited = vec![];
-            let mut visitor_fn = |node: &Node| {
+            let mut visitor_fn = |node: &Node, _: &_| {
                 visited.push(node.clone());
```

Comment:
I'm wondering if it makes sense to assert for proper paths on the way too as in the original PR's tests.

Reply:
Aah good point.

Reply:
I'll just cherry-pick your tests.